### PR TITLE
Fix a crash while placing random castles in the Editor

### PR DESCRIPTION
**Pull request by ihhub**
relates to #6845

### CODE REVIEW

File: src/fheroes2/maps/map_format_helper.cpp [hunk 1]
```diff
@@ -227,7 +227,7 @@ namespace Maps
         // Towns can be neutral so 1 more color for them.
         std::array<bool, mainColors + 1> townColorsPresent{ false };
         std::array<uint8_t, mainColors> heroRacesPresent{ 0 };
-        std::array<uint8_t, mainColors> townRacesPresent{ 0 };
+        std::array<uint8_t, mainColors + 1> townRacesPresent{ 0 };
```
Comment (Districh-ru) [inline src/fheroes2/maps/map_format_helper.cpp:230]: Hi, @ihhub , should we do the same for random race heroes (`heroRacesPresent`)?

Comment (ihhub) [reply src/fheroes2/maps/map_format_helper.cpp:230]: Hi @Districh-ru , we don't have neutral heroes :) So there is no need of doing this.

Comment (Districh-ru) [reply src/fheroes2/maps/map_format_helper.cpp:230]: Sure, heroes can not be neutral. But Im talking about random ones. :)
изображение

Comment (ihhub) [reply src/fheroes2/maps/map_format_helper.cpp:230]: Random heroes have color. The crash was for neutral castles that have no color.

Comment (Districh-ru) [reply src/fheroes2/maps/map_format_helper.cpp:230]: Ow, sorry. :(
I was a little confused: the PR name has "Random", the variable name has "Races" and nothing about color. And the number of main colors and races are the same in the game and can be used for the length of array for races and for colors...
So I treated the `townRacesPresent` as count of towns for race, but should as `townRacesPresentForColor`.

**Review by Districh-ru — APPROVED** [review]

**Pull request merged**